pkg/iac/terraform/state/backend/cloud_reader.go

```diff
@@ -0,0 +1,63 @@
+package backend
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const BackendKeyCloud = "tfcloud"
+const TerraformCloudAPI = "https://app.terraform.io/api/v2"
+
+type TFCloudAttributes struct {
+	HostedStateDownloadUrl string `json:"hosted-state-download-url"`
+}
+
+type TFCloudData struct {
+	Attributes TFCloudAttributes `json:"attributes"`
+}
+
+type TFCloudBody struct {
+	Data TFCloudData `json:"data"`
+}
+
+func NewCloudReader(workspaceId string, opts *Options) (*HTTPBackend, error) {
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/workspaces/%s/current-state-version", TerraformCloudAPI, workspaceId), nil)
+
+	req.Header.Add("Content-Type", "application/vnd.api+json")
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", opts.TerraformCloudToken))
+
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Isn't `req` going to be `nil` if there's an error here? Error handling should be right next to the expression IMO